core/src/main/java/com/dtp/core/thread/DtpExecutor.java

```diff
@@ -156,7 +156,7 @@ protected void beforeExecute(Thread t, Runnable r) {
         super.beforeExecute(t, r);
         DtpRunnable runnable = (DtpRunnable) r;
         runnable.cancelTimeoutCheckTask(QUEUE_TIMEOUT);
```

Comment:
If the cancellation is done in beforeExecute(), there may be an issue if the rejection policy is triggered before reaching this point, as it will not be executed.